Crashing ecl when loading "babel" with 2.20

Hi, ecl is crashing when trying to install "babel" through emacs/slime. It works when  ecl is run through the command line (no swank). See [earlier issue on ecl repo](https://gitlab.com/embeddable-common-lisp/ecl/issues/422#note_50706671) for information.

Hi, sorry for the short reply last. Was in a rush.

I am still getting the error though. My setup is as follows:

- Spacemacs (develop branch).
- I deleted .slime and also turned off "common-lisp" layer from spacemacs (so no elpa slime).
- I installed ecl through homebrew this time (usually I use roswell but same error there).
- I put in the configuration you gave and (setq inferior-lisp-program "ecl"). It recompiled ecl/slime when it started.
  I also saw that it was loading the correct slime from the paths during compilation.
- I had to install quicklisp again. did this and (ql:quickload :babel).

The output is (only last part):

```common-lisp
;;; Emitting code for DECODER.
;;; Finished compiling /Users/izzy/quicklisp/dists/quicklisp/software/babel-20170630-git/src/enc-unicode.lisp.
;;;
;;;
;;; Compiling /Users/izzy/quicklisp/dists/quicklisp/software/babel-20170630-git/src/enc-cp1251.lisp.
;;; OPTIMIZE levels: Safety=2, Space=0, Speed=3, Debug=3
;;;
;;; End of Pass 1.
;;; Emitting code for DECODER.
;;; Emitting code for ENCODER.
;;; Finished compiling /Users/izzy/quicklisp/dists/quicklisp/software/babel-20170630-git/src/enc-cp1251.lisp.
;;;
;;;
;;; Compiling /Users/izzy/quicklisp/dists/quicklisp/software/babel-20170630-git/src/enc-cp1252.lisp.
;;; OPTIMIZE levels: Safety=2, Space=0, Speed=3, Debug=3
;;;
;;; End of Pass 1.
;;; Emitting code for DECODER.
;;; Emitting code for ENCODER.
;;; Finished compiling /Users/izzy/quicklisp/dists/quicklisp/software/babel-20170630-git/src/enc-cp1252.lisp.
;;;
;;;
;;; Compiling /Users/izzy/quicklisp/dists/quicklisp/software/babel-20170630-git/src/jpn-table.lisp.
;;; OPTIMIZE levels: Safety=2, Space=0, Speed=3, Debug=3
;;;
;;; Compiling (DEFVAR *CP932-ONLY* ...).
;;; Compiling (DEFVAR *EUCJP-ONLY* ...).
;;; Compiling (DEFVAR *EUCJP* ...).
;;; End of Pass 1.
;;; Finished compiling /Users/izzy/quicklisp/dists/quicklisp/software/babel-20170630-git/src/jpn-table.lisp.
;;;
```

Then I get the good old "inferior lisp unexpected closed connection" or something like this in the mini buffer.

The last of theoutput of the inferior-buffer:

```common-lisp
10 warnings generated.
.cache/common-lisp/ecl-16.1.3-unknown-macosx-x64/Users/izzy/quicklisp/dists/quicklisp/software/alexandria-20170830-git/types.c:158:49: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
    value0 = cl_list(3, ECL_SYM("INTEGER",437), ecl_make_fixnum(-2305843009213693952), ecl_make_fixnum(-1));
                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/Cellar/ecl/16.1.3_1/include/ecl/object.h:131:62: note: expanded from macro 'ecl_make_fixnum'
#define ecl_make_fixnum(n)      ((cl_object)(((cl_fixnum)(n) << 2) | t_fixnum))
                                              ~~~~~~~~~~~~~~ ^
.cache/common-lisp/ecl-16.1.3-unknown-macosx-x64/Users/izzy/quicklisp/dists/quicklisp/software/alexandria-20170830-git/types.c:158:88: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
    value0 = cl_list(3, ECL_SYM("INTEGER",437), ecl_make_fixnum(-2305843009213693952), ecl_make_fixnum(-1));
                                                                                       ^~~~~~~~~~~~~~~~~~~
/usr/local/Cellar/ecl/16.1.3_1/include/ecl/object.h:131:62: note: expanded from macro 'ecl_make_fixnum'
#define ecl_make_fixnum(n)      ((cl_object)(((cl_fixnum)(n) << 2) | t_fixnum))
                                              ~~~~~~~~~~~~~~ ^
.cache/common-lisp/ecl-16.1.3-unknown-macosx-x64/Users/izzy/quicklisp/dists/quicklisp/software/alexandria-20170830-git/types.c:188:49: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
    value0 = cl_list(3, ECL_SYM("INTEGER",437), ecl_make_fixnum(-2305843009213693952), ecl_make_fixnum(0));
                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/Cellar/ecl/16.1.3_1/include/ecl/object.h:131:62: note: expanded from macro 'ecl_make_fixnum'
#define ecl_make_fixnum(n)      ((cl_object)(((cl_fixnum)(n) << 2) | t_fixnum))
                                              ~~~~~~~~~~~~~~ ^
.cache/common-lisp/ecl-16.1.3-unknown-macosx-x64/Users/izzy/quicklisp/dists/quicklisp/software/alexandria-20170830-git/types.c:406:66: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
    value0 = cl_list(3, ECL_SYM("INTEGER",437), ECL_SYM("*",18), ecl_make_fixnum(-1));
                                                                 ^~~~~~~~~~~~~~~~~~~
/usr/local/Cellar/ecl/16.1.3_1/include/ecl/object.h:131:62: note: expanded from macro 'ecl_make_fixnum'
#define ecl_make_fixnum(n)      ((cl_object)(((cl_fixnum)(n) << 2) | t_fixnum))
                                              ~~~~~~~~~~~~~~ ^
4 warnings generated.

Process inferior-lisp illegal instruction: 4

```

Is nobody else having this issue?

I am guessing this error is pretty low level and I want to mention again that I did migrate a lot of OS tools (sed/awk/coreutils etc) to GNU tools even though I am on macOS. Could this be the problem? I have tried to undo it but I could imagine that something might still be stuck.

Is there a way I can get a more useful error report than illegal instruction 4? And what does bisection over slime mean?

--- a/swank/ecl.lisp
+++ b/swank/ecl.lisp
@@ -1057,7 +1057,9 @@
 
 ;;;; Weak datastructures
 
-#+ecl-weak-hash
+;;; XXX: this should work but causes SLIME REPL hang at some point of time. May
+;;; be ECL or SLIME bug - disabling for now.
+#+(and ecl-weak-hash (or))
 (progn
   (defimplementation make-weak-key-hash-table (&rest args)
     (apply #'make-hash-table :weakness :key args))

